File: go-client-sdk/pkg/models/operations/getbatchcrmuser.go
package operations

import (
	"github.com/speakeasy-sdks/orb-sdks/go-client-sdk/pkg/models/shared"
)

type GetBatchCrmUserQueryParams struct {
	AccessToken string  `queryParam:"style=form,explode=true,name=accessToken"`
	AllFields   *bool   `queryParam:"style=form,explode=true,name=allFields"`
	Ids         *string `queryParam:"style=form,explode=true,name=ids"`
}

type GetBatchCrmUserResponseBody struct {
	InvalidIds []string      `json:"invalidIds,omitempty"`
	Users      []shared.User `json:"users,omitempty"`
}

type GetBatchCrmUserRequest struct {
	QueryParams GetBatchCrmUserQueryParams
}

type GetBatchCrmUserResponse struct {
	ContentType  string
	StatusCode   int64
	ResponseBody *GetBatchCrmUserResponseBody
}

File: typescript-client-sdk/src/sdk/models/operations/getbatchcrmuser.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetBatchCrmUserQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=accessToken" })
  accessToken: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=allFields" })
  allFields?: boolean;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=ids" })
  ids?: string;
}


export class GetBatchCrmUserResponseBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=invalidIds" })
  invalidIds?: string[];

  @SpeakeasyMetadata({ data: "json, name=users", elemType: shared.User })
  users?: shared.User[];
}


export class GetBatchCrmUserRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: GetBatchCrmUserQueryParams;
}


export class GetBatchCrmUserResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  responseBody?: GetBatchCrmUserResponseBody;
}
